Formula/rpm.rb

```diff
@@ -27,51 +13,40 @@ class Rpm < Formula
     sha256 "3a276b6a3f7273f8f88effbb057e8d65fc53fa9e812054951bfdd5ce618f54ac" => :yosemite
   end
 
-  depends_on "rpm2cpio" => :build
+  depends_on "beecrypt"
   depends_on "berkeley-db"
   depends_on "libmagic"
+  depends_on "libarchive"
   depends_on "popt"
-  depends_on "libtasn1"
+  depends_on "gnupg"
```

Comment:
This should probably be `depends_on :gpg => :run`, but are you sure this isn't something people can just choose to install if they want to use it but otherwise not install?

Reply:
This could be made optional, yes.

Reply:
Optional would trigger a source build if people used the option, so probably we can just leave it out unless it's actually needed during the build in order to function?

Reply:
We can leave it out, as it's not needed for the build. If it's not there, the `%__gpg` macro defaults to `/usr/bin/gpg2`, which would be wrong, though. It could be set to build-time only so that the path would be correct?

Reply:
We can't just set the path with configure?

Reply:
Maybe. Worst case, we could just substitute it in via a replace.

Reply:
```
Josephs-MacBook-Pro:Formula joe$ brew deps --tree --include-build gnupg
gnupg
├── pkg-config
├── npth
├── gnutls
│   ├── xz
│   ├── pkg-config
│   ├── libtasn1
│   ├── gmp
│   │   └── xz
│   ├── nettle
│   │   └── gmp
│   │       └── xz
│   ├── libunistring
│   │   └── xz
│   └── p11-kit
│       ├── libffi
│       └── pkg-config
├── libgpg-error
├── libgcrypt
│   └── libgpg-error
├── libksba
│   └── libgpg-error
├── libassuan
│   └── libgpg-error
├── pinentry
│   ├── pkg-config
│   ├── libgpg-error
│   └── libassuan
│       └── libgpg-error
├── gettext
│   ├── xz
│   └── libxml2
│       ├── autoconf
│       ├── automake
│       │   ├── xz
│       │   └── autoconf
│       ├── libtool
│       │   └── xz
│       └── pkg-config
├── adns
└── libusb
```
That seems a bit over the top just to get a path right. Surely there's some other way …

Reply:
I'll just substitute in the path we know it'll be in. `/usr/local/bin/gpg`.

Reply:
Probably want
```
Gpg.gpg || Gpg.gpg2 || HOMEBREW_PREFIX/"bin/gpg"
```
@DomT4 unless you have a different suggestion?

Reply:
(also would need to be in `post_install` if we do it that way)

Reply:
Well, at this point, it'd only be `/usr/local/bin/gpg` since upstream switched gpg2 to put out binaries with that name.

Reply:
@Conan-Kudo yes in most cases `Gpg.gpg || Gpg.gpg2 || HOMEBREW_PREFIX/"bin/gpg"` would return `/usr/local/bin/gpg` :) See https://github.com/Homebrew/brew/blob/master/Library/Homebrew/gpg.rb

Reply:
> Gpg.gpg || Gpg.gpg2 || HOMEBREW_PREFIX/"bin/gpg"

This is accurate, for now. At some point I'm going to combine the syntax to simply be able to call `Gpg.executable` and get a path back. It's on my list.

Reply:
But yeah, either way if you're using a dep it should be some form of `:gpg` rather than a hard `"gnupg"`.

Reply:
What is this `:gpg` dep?

Reply:
Yeah, I'm thinking we can avoid the dep as long as it's a string replace in `post_install` with the `HOMEBREW_PREFIX/"bin/gpg"` fallback.

Reply:
GPG is primarily used to sign and verify RPMs, so basically that functionality doesn't work when GnuPG isn't installed.

Reply:
> What is this :gpg dep?

It's a special requirement. If you install the formula from source and have a `gpg` or `gpg2` executable available locally that equals or is newer than GnuPG 2.0 the dep is satisfied. If you pour from bottle or the dep isn't satisfied Homebrew's `gnupg` formula gets pulled in for you.

> Yeah, I'm thinking we can avoid the dep as long as it's a string replace in post_install with the HOMEBREW_PREFIX/"bin/gpg" fallback.

Yeah. Unless this blows up in practice I don't have an issue with it being tried at least.

Reply:
How do I use this?

Reply:
The requirement or the `post_install` idea?

Reply:
The requirement. I think I know how to use the post_install idea.

Reply:
The requirement is usually
```
  depends_on :gpg => :run
``

The post_install idea would be:
```
  def post_install
    gpg = Gpg.gpg || Gpg.gpg2 || HOMEBREW_PREFIX/"bin/gpg"
    inreplace "/whatever/file/it's/in/", "/usr/bin/gpg2", gpg
  end
```